src/cdk/table/sticky-styler.ts

```diff
@@ -107,6 +115,9 @@ export class StickyStyler {
     const startPositions = this._getStickyStartColumnPositions(cellWidths, stickyStartStates);
     const endPositions = this._getStickyEndColumnPositions(cellWidths, stickyEndStates);
 
+    const lastStickyStart = stickyStartStates.lastIndexOf(true);
+    const firstStickyEnd = stickyEndStates.indexOf(true);
```

Comment:
Do we need some logic to verify that the first and last sticky indexes are different?

Reply:
I think as long as the same column is not specified as sticky and stickyEnd, this will be fine. I think that if both of those were set, getting both of these CSS classes would be the least of the odd behaviors to result.